### ISSUE TITLE
S7 Empty on Exit issue

### ISSUE THREAD

**Issue by Harbinger-365** [AppleWin/AppleWin#790]
Hi guys,

Thanks again for adding in both the -s7-empty-on-exit and the -conf command line options. I have noticed however that when using the -conf flag -s7-empty-on-exit no longer works. In what might be a related issue I dont think Starting Directory, or HDV Starting Directory are updating in the ini file when you load an image via -d1, -d2, -h1, etc which means that pressing f11 to save or f12 to load will look in the folder of the last image you manually loaded instead of the folder of the image mounted via -d1. I think it also effects which folder is shown if you press f3\f4 to load another disk.

Hope thats all clear, if not please let me know and ill try to explain it different.

**Comment by tomcw** [comment]
Thanks for raising this. I will take a look and report back.

**Comment by tomcw** [comment]
>I have noticed however that when using the -conf flag -s7-empty-on-exit no longer works.

This regressed at 1.29.7.0 (Dec'19) - when I added '-s5 diskii'.
NB. -s7-empty-on-exit was added at 1.29.3.0 (Oct'19).

**Comment by tomcw** [comment]
>In what might be a related issue I dont think Starting Directory, or HDV Starting Directory are updating in the ini file when you load an image via -d1, -d2, -h1, etc

I can reproduce this issue using -d1, -d2 (but not -h1, -h2).

**Comment by Harbinger-365** [comment]
Thanks for looking into it Tom

**Comment by tomcw** [comment]
Added tests for these regressions: https://github.com/AppleWin/AppleWin-Test/commit/b7782867f41e02417945f30dbc615aaa93eb9869

**Comment by tomcw** [comment]
@Harbinger-365 - please try this new AppleWin 1.29.13.0, [here](https://github.com/AppleWin/AppleWin/releases/tag/v1.29.13.0), and let me know if it resolves the issues you reported here.

**Comment by Harbinger-365** [comment]
Hi Tom, Looks like that version has fixed the issues. Thanks!

While looking into this I think I found whats causing #691 as well. I think its something to do with when the "starting directory" line is being updated when using the -d1\-d2 command line switch. When you load applewin and press f11 or f12 it will open whatever folder was in the "starting directory" line before applewin loaded, not the folder of the game currently loaded. This makes me think that whatever is behind f11\f12 is looking at the "starting directory" line before it has been updated to the current path. I currently get around this by loading a game, quiting, and loading it again in my bat file. Its even worse when using the -h1\-h2 command line switch as it never updates the "starting directory" line at all due to using "HDV Starting Directory" so even on a subsequent load f11\f12 will not load the folder of the current game.

Again if that doesnt make sense let me know and ill try to explain what I mean another way.
